### PR TITLE
P2P documentation

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds the P2P documentation.

### CODE REVIEW

File: docs/agent.rst [hunk 1]
```diff
@@ -105,3 +105,48 @@ Once it is built successfully, you just need to launch it executing one of the f
 * The ``-v,--verbose <level[0-6]>`` option sets log level from less to more verbose, in  level 0 the logger is off.
 * ``-m,--middleware <middleware-impl>``: set the middleware implementation to use. There are two: DDS (specified by the XRCE standard) and Centralized (topic are managed by the Agent similarly MQTT).
 * The ``--p2p <port>`` option enables P2P communication. Centralized middleware is necessary for this option.
+
+Middleware Abstraction Layer
+----------------------------
+
+The Middleware Abstraction Layer is an interface whose purpose is isolated the XRCE core from the middleware, as well as, to allow providing multiple middleware implementations.
```
Comment (BorjaOuterelo) [inline docs/agent.rst]:
```suggestion
The Middleware Abstraction Layer is an interface whose purpose is to isolate the XRCE core from the middleware, as well as, to allow providing multiple middleware implementations.
```

File: docs/agent.rst [hunk 2]
```diff
@@ -105,3 +105,48 @@ Once it is built successfully, you just need to launch it executing one of the f
...
+The interface has a set of pure virtual functions, which are called by the `ProxyClient` each time a *Client* requests for creating/deleting an entity or write/read data.
+
+.. image:: images/middleware_abstraction_layer.svg
+
+For the moment, the *Agent* accounts with two middleware implementations: *FastMiddleware* and *CedMiddleware*.
```
Comment (BorjaOuterelo) [inline docs/agent.rst]:
```suggestion
For the moment, the *Agent* counts with two middleware implementations: *FastMiddleware* and *CedMiddleware*.
```

File: docs/agent.rst [hunk 3]
```diff
@@ -105,3 +105,48 @@ Once it is built successfully, you just need to launch it executing one of the f
...
+
+FastMiddleware
+^^^^^^^^^^^^^^
+
+The *FastMiddleware* uses *eProsima Fast RTPS*, a C++ implementation of the RTPS (Real Time Publish Subscribe) protocol.
+This middleware allows *Client* to produce and consume data in the DDS Global Data Space, and consequently in the ROS 2 system.
+In that case, the *Agent* has the default behaviour describes in the DDS-XRCE standard, that is, for each DDS-XRCE entity and DDS proxy entity is created, and the writing/reading action produces a publishing/subscribing operation in the DDS world.
```
Comment (BorjaOuterelo) [inline docs/agent.rst]:
```suggestion
In that case, the *Agent* has the default behaviour as described in the DDS-XRCE standard, that is, for each DDS-XRCE entity a DDS proxy entity is created, and the writing/reading action produces a publishing/subscribing operation in the DDS world.
```

File: docs/agent.rst [hunk 4]
```diff
@@ -105,3 +105,48 @@ Once it is built successfully, you just need to launch it executing one of the f
...
+
+.. _ced_middleware_label:
+
+CedMiddleware
+^^^^^^^^^^^^^
+
+The *CedMiddleware* (Centralized Middleware) works similar to MQTT, that is, the *Agent* acts as a MQTT broker:
```
Comment (BorjaOuterelo) [inline docs/agent.rst]:
```suggestion
The *CedMiddleware* (Centralized Middleware) works similar to MQTT, that is, the *Agent* acts as a broker:
```

File: docs/agent.rst (hunk 2 above)
Comment (julianbermudez) [reply docs/agent.rst]: Fixed!

File: docs/agent.rst (hunk 3 above)
Comment (julianbermudez) [reply docs/agent.rst]: Fixed!

File: docs/agent.rst (hunk 4 above)
Comment (julianbermudez) [reply docs/agent.rst]: Fixed!

**Pull request merged**